distributed/utils.py

```diff
@@ -1757,3 +1761,23 @@ def remaining(self) -> float | None:
     def expired(self) -> bool:
         """Whether the deadline has already expired"""
         return self.remaining == 0
+
+
+class RateLimiterFilter(logging.Filter):
+    """A Logging filter that ensures a matching message is emitted at most every
+    `rate` seconds"""
+
+    def __init__(self, name, pattern, rate="10s"):
+        self.name = name
+        self.rate = parse_timedelta(rate)
+        self.pattern = re.compile(pattern)
+        self._last_seen = -1.0
+
+    def filter(self, record):
+        now = monotonic()
+        if self.pattern.match(record.msg):
+            old = self._last_seen
+            self._last_seen = now
+            if now - old > self.rate:
```

Comment:
```suggestion
        if self.pattern.match(record.msg):
            now = monotonic()
            if now - self._last_seen > self.rate:
                self._last_seen = now
```
Otherwise it will suppress *all* messages, *forever*, as long as they are emitted within 10s of each other.